core/src/main/scala/org/apache/spark/deploy/DependencyUtils.scala

```diff
@@ -65,7 +65,7 @@ private[deploy] object DependencyUtils extends Logging {
       .map {
         resolveGlobPaths(_, hadoopConf)
           .split(",")
-          .filterNot(_.contains(userJar.split("/").last))
+          .filterNot(_.contains(userJar.split(System.getProperty("file.separator")).last))
```

Comment:
`File.separator` is better

Reply:
+1

Reply:
I have updated to use File.separator instead of System.getProperty("file.separator").